File: Go/Base64Encoding.go
package main

import (
  "fmt"
  "strings"
  "net/http"
  "io/ioutil"
  "encoding/base64"
)

// Basic authentication example
func main() {

  // Customer key
  customerKey := "Your customer key"
  // Customer secret
  customerSecret := "Your customer secret"

  // Use base64 encoding
  plainCredentials := customerKey + ":" + customerSecret
  base64Credentials := base64.StdEncoding.EncodeToString([]byte(plainCredentials))

  url := "https://api.agora.io/dev/v2/project/<Your App ID>/rtm/users/userA/peer_messages?wait_for_ack=true"
  method := "POST"

  payload := strings.NewReader(`{"destination": "userB",  "enable_offline_messaging": false,  "enable_historical_messaging": false,   "payload": "Hello"}`)

  client := &http.Client {
  }
  req, err := http.NewRequest(method, url, payload)

  if err != nil {
    fmt.Println(err)
    return
  }
  // Add Authorization header and content type
  req.Header.Add("Authorization", "Basic " + base64Credentials)
  req.Header.Add("Content-Type", "text/plain")

  // Send HTTP request
  res, err := client.Do(req)
  if err != nil {
    fmt.Println(err)
    return
  }
  defer res.Body.Close()

  body, err := ioutil.ReadAll(res.Body)
  if err != nil {
    fmt.Println(err)
    return
  }
  fmt.Println(string(body))
}

File: Python/base64encoding.py
# -- coding utf-8 --
# Python 3
# Basic authentication example
import base64
import http.client

# Your customer key
customer_key = "Your customer key"
# Your customer secret
customer_secret = "Your customer secret"

# Base64 encoding
credentials = customer_key + ":" + customer_secret
base64_credentials = base64.b64encode(credentials.encode("utf8"))

# Credential to add to the Authorization field
credential = base64_credentials.decode("utf8")

# Create connection with the base URL
conn = http.client.HTTPSConnection("api.agora.io")

# Specify the payload to send with the request
payload = "{\r\n    \"destination\": \"userB\",\r\n    \"enable_offline_messaging\": false,\r\n    \"enable_historical_messaging\": false,\r\n    \"payload\": \"Hello\"\r\n}\r\n\r\n"

# Header to send with the request
headers = {}
headers['Authorization'] = 'basic ' + credential
headers['Content-Type'] = 'text/plain'

# Send request with path
conn.request("POST", "/dev/v2/project/<your-App-ID>/rtm/users/userA/peer_messages?wait_for_ack=true", payload, headers)
res = conn.getresponse()
data = res.read()
print(data.decode("utf-8"))